### PR TITLE
docs: move flags not supported by Setup.hs, fixes #7859

### LINKED ISSUE #7859: --allow-newer with Setup.hs configure?
Docs list `--allow-newer` under flags that [`runhaskell Setup.hs configure` supports][1], but it doesn't seem to actually do anything. The only way I've found to break the bounds with `Setup.hs configure` is to use `--exact-configuration` and specify everything.

[1]: https://cabal.readthedocs.io/en/3.6/setup-commands.html#cmdoption-runhaskell-Setup.hs-configure-allow-newer

### PR DESCRIPTION
**Pull request by adamse**
---
Please include the following checklist in your PR:

* [ ] Patches conform to the [coding conventions](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#conventions).
* [ ] Any changes that could be relevant to users [have been recorded in the changelog](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#changelog).
* [ ] The documentation has been updated, if necessary.

Please also shortly describe how you tested your change. Bonus points for added tests!

### CODE REVIEW

File: doc/setup-commands.rst [hunk 1]
```diff
@@ -1030,66 +957,6 @@ Miscellaneous options
         $ cabal install --constraint="foo:bar:exe.baz == 1.0"
 
 .. option:: --preference=preference
```
Comment (Mikolaj) [inline doc/setup-commands.rst:959]: leftover?

File: doc/cabal-commands.rst [hunk 2]
```diff
@@ -51,6 +51,143 @@ is in place, which moves the old configuration to a ``cabal.project.local~``
 file, this feature can also be disabled by using the ``--disable-backup``
 flag.
 
+.. option:: --default-user-config=file
```
Comment (Mikolaj) [inline doc/cabal-commands.rst:54]: Here, we are in a subsection about `v2-configure`. Is there a section/subsection somewhere about commands common to all v2- commands?

Comment (adamse) [reply doc/cabal-commands.rst:54]: i can put them at the top section (https://cabal.readthedocs.io/en/3.6/cabal-commands.html)?

Comment (Mikolaj) [reply doc/cabal-commands.rst:54]: Sure. Or create a distinct subsection --- whichever you choose. Now that I think about it, I'm not sure these are common to all v2-commands. There may be exceptions. Are there? If so, a couple of sentences before the command listing starts should say that they are common to many v2-commands, but to determine which belongs to which, one should use `cabal help cmd`.

Comment (Mikolaj) [reply doc/cabal-commands.rst:54]: BTW, this links is better: https://cabal.readthedocs.io/en/latest

Comment (Mikolaj) [reply doc/cabal-commands.rst:54]: I mean, the reader of the docs needs to experiment. Yes, moving is enough, with a sentence that says something, but not too much --- this is what I'm agonizing about.

**Comment by Mikolaj** [comment]
I'm re-running the failing CI job.

File: doc/cabal-commands.rst (hunk 2 above)
Comment (Mikolaj) [reply doc/cabal-commands.rst:8]: Perhaps just tweak "describing the arguments and flags they accept"? E.g., "first describing the arguments and flags that are common to some or all of them"? That implies that args and flags in subsections are not common.

**Review by Mikolaj — APPROVED** [review]
Looks great. Thank you!

**Review by Kleidukos — APPROVED** [review]
LGTM. :)

**Pull request merged**